ports/esp32/portnativeglue.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Jim Mussared & Thorsten von Eicken
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// We need the MICROPY_PORT_FUN_TABLE #define from mpconfigport.h but need to exclude pulling in the
+// rest of the .h world because it spoils the phony "voidfun" type decl below. We #define some
+// XXX_H defines to prevent recursive inclusion. Alternatives would be to pass
+// -DMICROPY_PORT_FUN_TABLE on the gcc commandline or to build the mp_port_fun_table table in asm.
+#define _STDINT_H
+#define _NEWLIB_ALLOCA_H
+#define _SYS_TYPES_H
+#define _ROM_ETS_SYS_H_
+#define INC_FREERTOS_H
+#pragma GCC diagnostic ignored "-Wbuiltin-declaration-mismatch"
```

Comment:
To get Travis to build, remove this and instead have a rule in the Makefile to disable this error.